This is a hotfix for KDE 3.5.5 with CUPS 1.2.x.

Copied from http://bugs.kde.org/show_bug.cgi?id=115891:
Empty printer list fix patch 2006-11-01 18:06:33

--- kde.orig/kdeprint/cups/ipprequest.cpp
+++ kde.patched/kdeprint/cups/ipprequest.cpp
@@ -329,6 +329,10 @@
 		dumpRequest(request_, true);
 	}
 
+	/* No printers found */
+	if ( request_ && request_->request.status.status_code == 0x406 )
+		return true;
+
 	if (!request_ || request_->state == IPP_ERROR || (request_->request.status.status_code & 0x0F00))
 		return false;
 
--- kde.orig/kdeprint/cups/kmcupsmanager.cpp
+++ kde.patched/kdeprint/cups/kmcupsmanager.cpp
@@ -61,7 +61,7 @@
 #define ppdi18n(s)	i18n(QString::fromLocal8Bit(s).utf8())
 
 void extractMaticData(QString& buf, const QString& filename);
-QString printerURI(KMPrinter *p, bool useExistingURI = false);
+QString printerURI(KMPrinter *p, bool useExistingURI);
 QString downloadDriver(KMPrinter *p);
 
 static int trials = 5;
@@ -218,7 +218,7 @@
 	QString		uri;
 
 	req.setOperation(state);
-	uri = printerURI(p);
+	uri = printerURI(p, true);
 	req.addURI(IPP_TAG_OPERATION,"printer-uri",uri);
 	if (req.doRequest("/admin/"))
 		return true;
